packages/backend/src/services/InstanceConfigurationService/InstanceConfigurationService.ts

```diff
@@ -0,0 +1,290 @@
+                true, // Skip permission check
+            );
+
+            // Optional steps are performed at the end
+            if (setup.organization.emailDomain) {
+                this.logger.debug(
+                    `Initial setup: Whitelisting domain "${setup.organization.emailDomain}"`,
+                );
+                const emailDomains = [setup.organization.emailDomain];
+                // Validates input
+                const error = validateOrganizationEmailDomains(emailDomains);
+                if (error) {
+                    throw new ParameterError(error);
+                }
+                const allowedDomains: AllowedEmailDomains = {
+                    organizationUuid,
+                    emailDomains,
+                    role: OrganizationMemberRole.VIEWER,
+                    projects: [],
+                };
+                await this.organizationAllowedEmailDomainsModel.upsertAllowedEmailDomains(
+                    allowedDomains,
+                );
+
+                this.logger.info(
+                    `Initial setup: Whitelisted domain "${setup.organization.emailDomain}"`,
+                );
+            } else {
+                this.logger.info(
+                    `Initial setup: No whitelisted domain, skipping`,
+                );
+            }
+
+            if (setup.serviceAccount && this.serviceAccountModel) {
+                this.logger.debug(`Initial setup: creating service account`);
+                await this.serviceAccountModel.save(
+                    sessionUser,
+                    {
+                        organizationUuid,
+                        expiresAt: setup.serviceAccount.expirationTime,
+                        description: 'Initial setup service account',
+                        scopes: [ServiceAccountScope.ORG_ADMIN],
+                    },
+                    setup.serviceAccount.token,
+                );
+                this.logger.info(`Initial setup: service account created`);
+            } else {
+                this.logger.info(
+                    `Initial setup: No service account token provided, skipping`,
+                );
+            }
+
+            if (setup.apiKey) {
+                this.logger.debug(`Initial setup: creating API key`);
+
+                await this.personalAccessTokenModel.save(sessionUser, {
+                    expiresAt: setup.apiKey.expirationTime,
+                    description: 'Initial setup API token',
+                    autoGenerated: false,
+                    token: setup.apiKey.token,
+                });
+                this.logger.info(`Initial setup: API key created`);
+            } else {
+                this.logger.info(
+                    `Initial setup: No API key provided, skipping`,
+                );
+            }
+        } catch (error) {
+            this.logger.error(
+                `Initial setup: Error initializing project: ${error}`,
```

Comment:
If we catch an error down here, do we want to throw? Won't the app continue to initialize in App.js as if this was successful?